extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/client/HashicorpVaultClient.java

```diff
@@ -121,7 +121,7 @@ public Result<Void> doHealthCheck() {
     public Result<Boolean> isTokenRenewable() {
         var uri = settings.url()
                 .newBuilder()
-                .addPathSegment(TOKEN_LOOK_UP_SELF_PATH)
+                .addPathSegments(TOKEN_LOOK_UP_SELF_PATH)
```

Comment:
isn't there a way to justify this change through an automated test?

Reply:
I switched [HashicorpVaultClientIntegrationTest](https://github.com/eclipse-edc/Connector/blob/8f5bff24a80c950a6c24cb09af1cc5695eefb56e/extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/client/HashicorpVaultClientIntegrationTest.java#L42) from "vault/1.9.6" to "hashicorp/vault:1.17.3" (and higher) -> lookUpToken_whenTokenNotExpired_shouldSucceed will fail without this patch.

docker stopped to update "vault" from 1.14 on - the newer "vault" implementations all come from hashicorp as a verified publisher.

I know the license problem why many people froze the reference implementation, but its a kind of bug anway if the reference implementation interprets its own spec now more literally.

Reply:
@drcgjung ok, but in the changelog there's no version update, please change the docker image version accordingly, there shouldn't be any licensing issue.

Reply:
For a bit of context: HashiCorp changed their licensing in August 2023, mainly to prevent companies from offering competing services that are based on Vault (and/or other HashiCorp offerings). 

Since the EDC project neither sells nor embeds HashiCorp community software, nor does it offer a product competitive to a HashiCorp product, I think we are in the clear. 

However, if there is a breaking API change between the freely available version (`1.9.6`) and the "commercial" version (`1.17.3`), that cannot be solved with a separate code path in the vault client, then this becomes a topic for the EDC Technical Committee how to proceed. 

Are you saying that there is a breaking API change between `1.9.6` and `1.17.3`?

Reply:
@paullatzelsperger we already have libraries that deal with proprietary software (see all the technology repos), we are not distributing hashicorp code.
License wise I see no issue here.
 
If a new version brought a breaking change I think we should provide either two versions of the module or a config to being able to make the EDC being able to interact with both "pre" and "post" breaking change, until the "pre" will get out of support.

Reply:
> @drcgjung Are you saying that there is a breaking API change between `1.9.6` and `1.17.3`?

No, I am saying that AFAICS from version 1.17.3 on, the vault API does not tolerate/interpret url-encoded slashes in path prefixes as normal slashes anymore (which is IMHO quite a difference from the W3C spec perspective). So the older versions seems to have been more "tolerant" wrt the interpretation of the same API.

My patch works with ALL versions, only the integration test currently needs to select a single reference - maybe it could be subclassed/doubled to test integration with both versions?

Reply:
@ndr-brt my point exactly. _We (=EDC)_ are in the clear, legally. My point is that we can't really recommend publicly to use vault `1.17.3`, because that is not strictly OSS anymore and that would be a problem from an EF perspective I think. Yes, we have implementations for other non-free software, but there always has to be an OSS alternative.

I guess a good middle ground would be to test against `1.17.3` AND the last-known-free version `1.9.6`, as @drcgjung suggested. That way we can guarantee that our code works with "an open-source variant".

Reply:
> Just to be sure and to guard against regression, a good middle ground would be to test against `1.17.3` AND the last-known-free version `1.9.6`, as @drcgjung suggested. That way we can guarantee that our code works with "an open-source variant".

definitely agree

Reply:
> > Just to be sure and to guard against regression, a good middle ground would be to test against `1.17.3` AND the last-known-free version `1.9.6`, as @drcgjung suggested. That way we can guarantee that our code works with "an open-source variant".
> 
> definitely agree

second integration test implemented.